interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt

```diff
@@ -648,34 +648,32 @@ private fun KmProperty.toPropertySpec(
   val returnTypeName = returnType.toTypeName(typeParamResolver)
   val mutableAnnotations = mutableListOf<AnnotationSpec>()
   if (containerData != null && propertyData != null) {
-    if (hasGetter) {
```

Comment:
Don't we need this check anymore?

Reply:
Seems like it was already likely redundant before. The nullable signature safe call does the same thing.

Reply:
Yes, it was redundant. This is the errors from building with 0.9.0 before any changes:
```
> Task :interop:kotlinx-metadata:compileKotlin
e: file:///usr/local/google/home/sgjesse/prj/kotlinpoet/interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/KotlinPoetMetadata.kt:88:38 Using 'read(Metadata): KotlinClassMetadata' is an error. read() throws an error if metadata version is too high. Use either readStrict() if you want to retain this behavior, or readLenient() if you want to try to read newer metadata.
e: file:///usr/local/google/home/sgjesse/prj/kotlinpoet/interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt:651:9 Using 'hasGetter: Boolean' is an error. Kotlin properties always have getters
e: file:///usr/local/google/home/sgjesse/prj/kotlinpoet/interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt:677:9 Using 'hasSetter: Boolean' is an error. Check .setter for nullability instead
e: file:///usr/local/google/home/sgjesse/prj/kotlinpoet/interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt:780:11 Using 'hasGetter: Boolean' is an error. Kotlin properties always have getters
e: file:///usr/local/google/home/sgjesse/prj/kotlinpoet/interop/kotlinx-metadata/src/main/kotlin/com/squareup/kotlinpoet/metadata/specs/KotlinPoetMetadataSpecs.kt:788:11 Using 'hasSetter: Boolean' is an error. Check .setter for nullability instead
```